### DISCUSSION TITLE
issue tso commands and why is a address space started each time?

### DISCUSSION THREAD

**Discussion by kraabrsg** [zowe/zowe-cli#1336] [Q&A] [answered]
Hi folks, i want to post a question about this in the zowe-cli slack channel, but unfortunatly i get an empty window somehow.

We are working on a vscode extension to issue tso, zosmf calls, etc.

When i issue a tso  status command,
`zowe zos-tso issue command "status"`
  from the cli, or from our vscode extension vis SDK, each time this command is sent a new job (assuming this also starts an seperate address space) is started

the tso profile is fairly standard
`zowe profiles create tso-profile profname --cs 695 --cp 1141 -a accont -l logonproc `

Can this behaviour be avoided somehow?

zowe --version
`6.37.0`

Thanks!

**Comment by t1m0thyj** [comment, marked as answer]
This is expected behavior for the [`zowe zos-tso issue command`](https://docs.zowe.org/stable/web_help/index.html?p=zowe_zos-tso_issue_command) command based on its description:
> Creates a TSO address space, issues a TSO command through the newly created address space, waits for the READY prompt to print the response, and terminates the TSO address space.

If this behavior is undesirable, you can use other commands in the [`zos-tso`](https://docs.zowe.org/stable/web_help/index.html?p=zowe_zos-tso) group to start/stop a single address space and send a command to the already existing address space. You can also do this using the `@zowe/zos-tso-for-zowe-sdk` package in a VS Code extension - see the "Demonstrate starting, pinging, and stopping a TSO address space" section [here](https://github.com/zowe/zowe-cli/tree/master/packages/zostso).
